src/emqtt.erl

```diff
@@ -1434,22 +1437,25 @@ handle_event(info, {Closed, _Sock}, connected, #state{ reconnect = Re} = State)
     when ?SOCK_CLOSED(Closed) andalso ?NEED_RECONNECT(Re) ->
     next_reconnect(State#state{socket = undefined});
 
+handle_event(info, {Closed, _Sock}, waiting_for_connack, #state{ reconnect = Re} = State)
+    when ?SOCK_CLOSED(Closed) andalso ?NEED_RECONNECT(Re) ->
+    next_reconnect(State#state{socket = undefined});
+
 handle_event(info, {Closed, Sock}, StateName, State)
     when Closed =:= tcp_closed; Closed =:= ssl_closed ->
-    ?LOG(debug, "socket_closed", #{event => Closed, state => StateName, sock => Sock,
+    ?LOG(error, "socket_closed", #{event => Closed, state => StateName, sock => Sock,
```

Comment:
there are quite a few places with this error silently ignored, do we really want to promote this one to error level ?

Reply:
This is for me to catch the event that caused stop. I will change it back to debug before merge.